### DISCUSSION TITLE
How DataGridTextColumn Binding?

### DISCUSSION THREAD

**Discussion by tsctsc6** [AvaloniaUI/Avalonia#16783] [Q&A] [answered]
Normally we would use the datagrid like this:
```
<DataGrid ItemsSource="{Binding People}">
    <DataGrid.Columns>
        <DataGridTextColumn Header="Name" Binding={Binding Name}/>
    </DataGrid.Columns>
</DataGrid>
```
How can DataGridTextColumn binding `People.Name` ? I create a custom control `MyControl`, and try to:
```
<MyControl ItemsSource="{Binding People}">
    <MyControl.DataGridColumns>
        <DataGridTextColumn Header="Name" Binding={Binding Name}/>
    </MyControl.DataGridColumns>
</MyControl>
```
However the compiler told me that "Can't find Property "Name" in ViewModel".

**Comment by thevortexcloud** [comment]
Does it work if you use a `ReflectionBinding` instead? If so you probably need to specify a datatype or just use a reflection binding. Reflection bindings however are discouraged though as you lose a lot of useful compile time functionality.

**Reply by tsctsc6** [reply]
Yes, it works!

I'm still curious though, why is it ok to just use `Binding` in a DataGrid? Maybe the `Binding` in `DataGrid` is actually `ReflectionBinding`, because I don't think compiler know what is the type of `People` in axaml.

**Reply by thevortexcloud** [reply]
As of v11 `Binding` by default is actually just functionally an alias for `CompiledBinding`. You can turn that off using some project setting, in which case it will act as a `ReflectionBinding` instead. A compiled binding needs an explicit data type to be set for it to work so the type can be checked at compile time. A reflection binding on the other hand has no compile time checks and lets you do some things that are impossible to do with a compiled binding. I personally always use explicit `ReflectionBinding`/`CompiledBinding` as it makes it far clearer what I am actually doing.

As for DataGrid itself, I am not sure. I am _guessing_ there is some attribute somewhere that makes it inherit type information from the parent.

**Comment by wujian0108** [comment, marked as answer]
```
<DataGrid 
            Margin="8"
            CanUserReorderColumns="True"
            CanUserResizeColumns="True"
            CanUserSortColumns="True"
            HeadersVisibility="All"
            IsReadOnly="True"
            ItemsSource="{Binding ProcessGridData}">
            <DataGrid.Columns>
                <DataGridTextColumn
                    Width="6*"
                    Header="ProcessName"
                    x:DataType="models:ProcessData"
                    Binding="{Binding Name}" />
           </DataGrid.Columns>
</DataGrid>
```

**Reply by tsctsc6** [reply]
It works too! Thanks.
